packages/core/src/routes/sso-connector/index.ts

```diff
@@ -199,4 +181,49 @@ export default function singleSignOnRoutes<T extends AuthedRouter>(...args: Rout
       return next();
     }
   );
+
+  /* Patch update a single sign on connector by id */
+  router.patch(
+    `${pathname}/:id`,
+    koaGuard({
+      params: z.object({ id: z.string().min(1) }),
+      body: ssoConnectorPatchGuard,
+      response: ssoConnectorWithProviderConfigGuard,
+      status: [200, 404, 422],
+    }),
+    async (ctx, next) => {
+      const { id } = ctx.guard.params;
+      const { body } = ctx.guard;
+
+      // Fetch the connector
+      const { providerName } = await ssoConnectors.findById(id);
+
+      // Return 422 if the connector provider is not supported
+      if (!isSupportedSsoProvider(providerName)) {
+        throw new RequestError({
+          code: 'connector.not_found',
+          type: providerName,
+          status: 422,
+        });
+      }
+
+      const { config, ...rest } = body;
+
+      // Validate the connector config if it's provided
+      const parsedConfig = parseConnectorConfig(providerName, config);
+
+      // Patch update the connector
+      const connector = await ssoConnectors.updateById(id, {
+        ...conditional(parsedConfig && { config: parsedConfig }),
+        ...rest,
```

Comment:
If the `rest` is an empty object, this may throw error because the sql update statement has empty data.

Reply:
updated